File: CShaprLib/Demos.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CShaprLib
{
    public class Demos : IDemos
    {
        /// <summary>
        ///  This is our C# LoadFile method
        /// </summary>
        /// <returns></returns>
        public List<string> LoadFile()
        {
            List<string> output = new List<string>();
            List<string> lines = File.ReadAllLines(@"C:\try\myfile.txt").ToList();
            for (int i = 0; i < lines.Count; i++)
            {
                //if( i % 2 == 0)
                    output.Add(lines[i]);
            }
            return output;
        }
        public void PrintFullName(string firstName, string lastName)
        {
            // You probably shouldn't do this in a class library
            Console.WriteLine($"{ firstName} { lastName }");
        }
    }
}

File: VBLib/Demos.vb
Imports System.IO

Public Class Demos
    Implements IDemos

    Public Function LoadFile() As List(Of String) Implements IDemos.LoadFile
        Dim output As List(Of String) = New List(Of String)
        Dim lines As List(Of String) = File.ReadAllLines("c:\try\myfile.txt").ToList()

        For index = 0 To lines.Count - 1
            If index Mod 2 = 0 Then
                output.Add(lines(index))
            End If
        Next
        Return output
    End Function

    Public Sub PrintFullName(firstName As String, lastName As String) Implements IDemos.PrintFullName
        Console.WriteLine($"{ firstName} {lastName}")
    End Sub
End Class
